arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowFieldType.java

```diff
@@ -93,6 +95,10 @@ public static ArrowFieldType of(ArrowType arrowType) {
       }
     case Date:
       return DATE;
+    case Time:
```

Comment:
@mihaibudiu The code provided by arrow does not seem to distinguish between precision
```
  public static enum ArrowTypeID {
    Null(Type.Null),
    Struct(Type.Struct_),
    List(Type.List),
    LargeList(Type.LargeList),
    FixedSizeList(Type.FixedSizeList),
    Union(Type.Union),
    Map(Type.Map),
    Int(Type.Int),
    FloatingPoint(Type.FloatingPoint),
    Utf8(Type.Utf8),
    LargeUtf8(Type.LargeUtf8),
    Binary(Type.Binary),
    LargeBinary(Type.LargeBinary),
    FixedSizeBinary(Type.FixedSizeBinary),
    Bool(Type.Bool),
    Decimal(Type.Decimal),
    Date(Type.Date),
    Time(Type.Time),
    Timestamp(Type.Timestamp),
    Interval(Type.Interval),
    Duration(Type.Duration),
    NONE(Type.NONE);
```

Reply:
The following is arrow's explanation of the time type
`Time is either a 32-bit or 64-bit signed integer type representing an elapsed time since midnight, stored in either of four units: seconds, milliseconds, microseconds or nanoseconds. The integer `bitWidth` depends on the `unit` and must be one of the following: * SECOND and MILLISECOND: 32 bits * MICROSECOND and NANOSECOND: 64 bits The allowed values are between 0 (inclusive) and 86400 (=24*60*60) seconds (exclusive), adjusted for the time unit (for example, up to 86400000 exclusive for the MILLISECOND unit). This definition doesn't allow for leap seconds. Time values from measurements with leap seconds will need to be corrected when ingesting into Arrow (for example by replacing the value 86400 with 86399).`

Reply:
The following is arrow's explanation of the above `Time` type

> Time is either a 32-bit or 64-bit signed integer type representing an elapsed time since midnight, stored in either of four units: seconds, milliseconds, microseconds or nanoseconds. The integer `bitWidth` depends on the `unit` and must be one of the following: * SECOND and MILLISECOND: 32 bits * MICROSECOND and NANOSECOND: 64 bits The allowed values are between 0 (inclusive) and 86400 (=24*60*60) seconds (exclusive), adjusted for the time unit (for example, up to 86400000 exclusive for the MILLISECOND unit). This definition doesn't allow for leap seconds. Time values from measurements with leap seconds will need to be corrected when ingesting into Arrow (for example by replacing the value 86400 with 86399).

Reply:
this is on the Arrow side, but most SQL types on the Calcite side have a precision, and some have a scale too. The question is about how the precision matters when generating arrow. Perhaps you can write at least one test using, let's say, TIMESTAMP(3) or CHAR(3) as a Calcite type.

Reply:
At present, it seems that the arrow adapter will translate the time type uniformly into time(0). This seems to be a bug. I will fix it.
@mihaibudiu I would like to ask a question about how calcite distinguishes the precision of the time type, such as Second == time(0), MilliSecond == time(3) and so on. I am not very clear about this aspect.

Reply:
Here is a quote from the postgres documentation: https://www.postgresql.org/docs/16/datatype-datetime.html

> time, timestamp, and interval accept an optional precision value p which specifies the number of fractional digits retained in the seconds field. By default, there is no explicit bound on precision. The allowed range of p is from 0 to 6.

Reply:
I am confirming the standards with people in the Arrow community